django/db/models/lookups.py

```diff
@@ -190,6 +198,49 @@ class FieldGetDbPrepValueIterableMixin(FieldGetDbPrepValueMixin):
     """
     get_db_prep_lookup_value_is_iterable = True
 
+    def get_prep_lookup(self):
+        prepared_values = []
+        if hasattr(self.rhs, '_prepare'):
+            # A subquery is like an iterable, but we shouldn't independently
+            # prepare its items
+            return self.rhs._prepare(self.lhs.output_field)
+        for rhs_value in self.rhs:
+            if isinstance(rhs_value, Combinable):
+                # A combinable expression will be handled by the database, but
+                # can co-exist alongside real values
+                pass
+            elif self.prepare_rhs and hasattr(self.lhs.output_field, 'get_prep_value'):
+                rhs_value = self.lhs.output_field.get_prep_value(rhs_value)
+            prepared_values.append(rhs_value)
+        return prepared_values
+
+    def process_rhs(self, compiler, connection):
+        if self.rhs_is_direct_value():
+            # rhs should be an iterable of 2 values, we use batch_process_rhs
```

Comment:
Couldn't there be more than 2 values for `__in` queries, or is this code not triggered for `__in`?